app/commands/cleanup/CleanupExerciseConfigs.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Console;
+
+use App\Model\Entity\Assignment;
+use App\Model\Entity\Exercise;
+use DateTime;
+use Doctrine\DBAL\Connection;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * A console command that removes exercise configs (all of them including limits) that are not associated with any
+ * exercise or assignment and were created before more than a given amount of days
+ */
+class CleanupExerciseConfigs extends Command {
+
+  /** @var EntityRepository */
+  private $exercises;
+
+  /** @var EntityRepository */
+  private $assignments;
+
+  /** @var EntityManager */
+  private $entityManager;
+
+  public function __construct(EntityManager $entityManager) {
+    parent::__construct();
+    $this->exercises = $entityManager->getRepository(Exercise::class); // even deleted exercises has to be found
```

Comment:
Why don't you inject those classes directly (as constructor arguments)? `$em->getRepository()` doesn't guarantee you'll get an `EntityRepository` - you can configure the repository class in the `@ORM\Entity` annotation. Maybe adding methods that bypass the deletedAt filter would be a better solution.

Reply:
Ok, I'll add new method to base soft delete repository